File: dart_cli/lib/structs/KeyMapping.dart

import "../ByteDataWrapper.dart";
import "../ioClass.dart";

class KeyMapping extends IoClass {
  late final int index;
  late final int keyLength;
  late final String keyName;

  KeyMapping(this.index, this.keyName) :
    keyLength = keyName.length;
  
  KeyMapping.readBytes(ByteDataWrapper bytes) {
    index = bytes.readUint16();
    keyLength = bytes.readUint8();
    keyName = bytes.readString(keyLength);
  }

  @override
  void writeBytes(ByteDataWrapper bytes) {
    bytes.writeUint16(index);
    bytes.writeUint8(keyLength);
    bytes.writeString(keyName);
  }

  @override
  int get size => 3 + keyLength;
}

File: python_lib/structs/KeyMapping.py
from FileWrapper import BinaryReader, BinaryWriter


class KeyMapping:
	index: int
	keyLength: int
	keyName: str

	def __init__(self, index: int, keyName: str):
		self.index = index
		self.keyName = keyName
		self.keyLength = len(keyName)

	@staticmethod
	def readBytes(bytes: BinaryReader):
		index = bytes.readUint16()
		keyLength = bytes.readUint8()
		keyName = bytes.readString(keyLength)
		return KeyMapping(index, keyName)

	def writeBytes(self, bytes: BinaryWriter):
		bytes.writeUint16(self.index)
		bytes.writeUint8(self.keyLength)
		bytes.writeString(self.keyName)

	@property
	def size(self) -> int:
		return 3 + self.keyLength
